templates/cases/case_read.haml

```diff
@@ -161,11 +161,16 @@
             - trans "Contact"
           .panel-body
             .container-fluid
-              .row
+              .row{ ng-repeat:"urn in contact.urns" }
+                .contact-field-label.col-sm-6
+                  [[ urn.scheme ]]
+                .contact-field-value.col-sm-6
+                  [[ urn.path ]]
+              .row{ ng-hide:"contact.urns.length > 0" }
                 .contact-field-label.col-sm-6
                   - trans "Name"
                 .contact-field-value.col-sm-6
-                  [[ caseObj.contact.name ]]
+                  [[ contact.name ]]
```

Comment:
For your usecase do you need to intentionally hide name even if it's set? If not I think it should be displayed even when a contact has urns - and I think be the top item.